### DISCUSSION TITLE
Regarding the possibility of Redash providing cross-data-source querying capabilities:

### DISCUSSION THREAD

**Discussion by Jengro777** [RedashCommunity/redash#40] [General]
I understand that there are many difficulties in performing cross-data-source querying on a single page, including:

1. Different data formats: Different data sources may have different data formats, which can make it difficult to combine and compare query results.

2. Access control: Cross-data-source queries require consideration of how to ensure user access control and data privacy across different data sources.

3. Communication between data sources: Communication between data sources may involve network delays and error handling.

4. Performance issues: Cross-data-source querying often involves processing and merging large amounts of data, which can impact performance and resource consumption.

Although cross-data-source querying is challenging, some BI tools offer this functionality. These tools typically connect to multiple data sources and provide a user-friendly interface to simplify cross-data-source querying, while also providing necessary access control and data privacy measures.

I would like to know the possibility of Redash providing multi-data-source querying, and thank you all for your input.

As far as I know, SmartBI supports cross-data source querying.
https://www.smartbi.com.cn/

**Comment by tinder-tder** [comment]
Have you tried setting up cached query results?
https://redash.io/help/user-guide/querying/query-results-data-source

there are limits (mainly memory) but it basically makes a sqlite bridge between results

**Reply by Jengro777** [reply]
Cool
